test/integration/test-custom-filename.js

```diff
@@ -35,6 +37,10 @@ var server = http.createServer(function(req, res) {
     assert.strictEqual(files['custom_filename'].name, options.filename, 'Expects custom filename');
     assert.strictEqual(files['custom_filename'].type, mime.lookup(knownFile), 'Expects original content-type');
 
+    assert('custom_filepath' in files);
+    assert.strictEqual(files['custom_filepath'].name, relativeFile.replace(/\\/g, '/'), 'Expects custom filename');
```

Comment:
Hah this is too funny. 😂

Anyway I think it's because I did `path.resolve` after the backslash replace. Seems on Windows `path.resolve` expects to still see the native OS separator. Makes sense, I suppose.

Should be fixed now. Also cleaned up some path concatenation in `text/common.js`, just to be on the safe side.